libs/zopengl/src/wrapper.zig

```diff
@@ -0,0 +1,2639 @@
+}
+
+// pub var getShaderInfoLog: *const fn (
+//     shader: Uint,
+//     bufSize: Sizei,
+//     length: [*c]Sizei,
+//     infoLog: [*c]Char,
+// ) callconv(.C) void = undefined;
+pub fn getShaderInfoLog(shader: Shader, buffer: []u8) ?[]const u8 {
+    assert(@bitCast(Uint, shader) > 0);
+    assert(buffer.len > 0);
+    var log_len: Sizei = 0;
+    bindings.getShaderInfoLog(
+        @bitCast(Uint, shader),
+        @intCast(Sizei, buffer.len),
+        &log_len,
+        @ptrCast([*c]Char, buffer.ptr),
+    );
+    return if (log_len > 0) buffer[0..@intCast(usize, log_len)] else null;
+}
+
+// pub var getShaderSource: *const fn (
+//     shader: Uint,
+//     bufSize: Sizei,
+//     length: [*c]Sizei,
+//     source: [*c]Char,
+// ) callconv(.C) void = undefined;
+
+// pub var getUniformLocation: *const fn (program: Uint, name: [*c]const Char) callconv(.C) Int = undefined;
+pub fn getUniformLocation(program: Program, name: [:0]const u8) ?UniformLocation {
+    assert(@bitCast(Uint, program) > 0);
+    const location = bindings.getUniformLocation(
+        @bitCast(Uint, program),
+        @ptrCast([*c]const Char, name.ptr),
+    );
+    return if (location >= 0) @bitCast(UniformLocation, location) else null;
+}
+
+// pub var getUniformfv: *const fn (program: Uint, location: Int, params: [*c]Float) callconv(.C) void = undefined;
+// pub var getUniformiv: *const fn (program: Uint, location: Int, params: [*c]Int) callconv(.C) void = undefined;
+// pub var getVertexAttribdv: *const fn (index: Uint, pname: Enum, params: [*c]Double) callconv(.C) void = undefined;
+// pub var getVertexAttribfv: *const fn (index: Uint, pname: Enum, params: [*c]Float) callconv(.C) void = undefined;
+// pub var getVertexAttribiv: *const fn (index: Uint, pname: Enum, params: [*c]Int) callconv(.C) void = undefined;
+// pub var getVertexAttribPointerv: *const fn (
+//     index: Uint,
+//     pname: Enum,
+//     pointer: [*c]?*anyopaque,
+// ) callconv(.C) void = undefined;
+// pub var isProgram: *const fn (program: Uint) callconv(.C) Boolean = undefined;
+// pub var isShader: *const fn (shader: Uint) callconv(.C) Boolean = undefined;
+
+// pub var linkProgram: *const fn (program: Uint) callconv(.C) void = undefined;
+pub fn linkProgram(program: Program) void {
+    assert(@bitCast(Uint, program) > 0);
+    bindings.linkProgram(@bitCast(Uint, program));
+}
+
+// pub var shaderSource: *const fn (
+//     shader: Uint,
+//     count: Sizei,
+//     string: [*c]const [*c]const Char,
+//     length: [*c]const Int,
+// ) callconv(.C) void = undefined;
+pub fn shaderSource(shader: Shader, src_ptrs: []const [*]const u8, src_lengths: []const usize) void {
+    assert(@bitCast(Uint, shader) > 0);
+    assert(src_ptrs.len > 0);
+    assert(src_ptrs.len == src_lengths.len);
+    bindings.shaderSource(
+        @bitCast(Uint, shader),
+        @intCast(Sizei, src_ptrs.len),
+        @ptrCast([*c]const [*c]const Char, src_ptrs),
+        @ptrCast([*c]const Int, src_lengths),
```

Comment:
This looks like a bug. A slice of `usize`s is passed in but we treat it like a pointer to some `Int`s, which has a different size?
I only tested this with one source ptr and len. Needs to be tested more.